balancer-js/src/modules/zaps/bbausd2-migrations/stables.ts

```diff
@@ -0,0 +1,246 @@
+import { StablePoolEncoder } from '@/pool-stable/encoder';
+import { ADDRESSES } from './addresses';
+import { Relayer } from '@/modules/relayer/relayer.module';
+import { ExitPoolRequest } from '@/types';
+import { BatchSwapStep, FundManagement, SwapType } from '@/modules/swaps/types';
+import { Interface } from '@ethersproject/abi';
+import { MaxUint256, MaxInt256 } from '@ethersproject/constants';
+// TODO - Ask Nico to update Typechain?
+import balancerRelayerAbi from '@/lib/abi/BalancerRelayer.json';
+import { BigNumber } from 'ethers';
+const balancerRelayerInterface = new Interface(balancerRelayerAbi);
+
+const SWAP_RESULT = Relayer.toChainedReference('0');
+const EXIT_RESULTS: BigNumber[] = [];
+
+export class StablesBuilder {
+  private addresses;
+
+  constructor(networkId: 1 | 5) {
+    this.addresses = ADDRESSES[networkId];
+  }
+
+  /**
+   * Builds migration call data.
+   * Migrates tokens from old stable to new stable phantom pools with the same underlying tokens.
+   * Tokens that are initially staked are re-staked at the end of migration. Non-staked are not.
+   *
+   * @param {string}                    userAddress User address.
+   * @param {{string, string, string}}  from Pool info being migrated from
+   * @param {{string, string, string}}  to Pool info being migrated to
+   * @param {string}                    bptIn Amount of BPT tokens to migrate.
+   * @param {string}                    minBptOut Minimum of expected BPT out ot the migration flow.
+   * @param {boolean}                   staked Indicates whether tokens are initially staked or not.
+   * @param {string[]}                  tokens Token addresses. Array must have the same length and order as tokens in pool being migrated from. Refer to [getPoolTokens](https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/interfaces/contracts/vault/IVault.sol#L334).
+   * @param {string}                    authorisation Encoded authorisation call.
+   * @returns Migration transaction request ready to send with signer.sendTransaction
+   */
+  calldata(
+    userAddress: string,
+    from: { id: string; address: string; gauge?: string },
+    to: { id: string; address: string; gauge?: string },
+    bptIn: string,
+    minBptOut: string = MaxInt256.toString(),
```

Comment:
I don't think we should make this optional as its important the user explicitly sets it to avoid issues.

Reply:
I agree - done ✅